GitCommands/Git/GitModule.cs

```diff
@@ -2883,7 +2883,8 @@ internal GitArgumentBuilder GetRefsCmd(bool tags, bool branches, bool noLocks)
                         ? (ArgumentString)"--no-optional-locks"
                         : default)
                 {
-                    { branches, "--dereference", "--tags" },
+                    { !branches, "--tags" },
```

Comment:
The logic here is weird, see #6044. If both tags and branches is set, other refs (like stash, notes, gerrit review, ...) are included. If only one of tags/branches is set, only those are included.
This change should be correct though.

This will include both the ref and the deref also if branches are requested at the same time.